collie/model/hybrid_matrix_factorization.py

```diff
@@ -466,20 +492,16 @@ def forward(self, users: torch.tensor, items: torch.tensor) -> torch.tensor:
                 + self.item_biases(items)
             )
         elif self.hparams.stage in ('metadata_only', 'all') and self.item_metadata is None:
-            # TODO: remove self.device and let lightning do it
-            user_metadata_output = self.user_metadata[users, :].to(self.device)
-            if self.user_metadata_layers is not None:
-                for metadata_nn_layer in self.user_metadata_layers:
-                    user_metadata_output = self.dropout(
-                        F.leaky_relu(
-                            metadata_nn_layer(user_metadata_output)
-                        )
-                    )
+            self._calculate_metadata_output(
+                metadata_type='user',
+                users=users,
+                items=items
+            )
 
             # TODO: make this matrix factorization instead of only a MLP
             combined_output = torch.cat((self.user_embeddings(users),
                                          self.item_embeddings(items),
-                                         user_metadata_output), 1)
+                                         self.user_metadata_output), 1)
```

Comment:
Can we adjust this order to be `user_metadata, user_embeddings, item_embeddings`?

Reply:
👍